client-js/app/elements/labrad-registry.html

```diff
@@ -59,8 +207,213 @@
+        });
+      }
+
+      else if (self.selectType==='key') {
+        p1 = self.socket.rename({path: self.path,key: self.selKey, newKey: self.$.newNameInput.value});
+        p1.then(function(resp) {
+            self.repopulateList(resp);
+        }, 
+        function(reason) {
+          self.handleError(reason); // Error!
+        });
+      }
+    },
+    doCopy: function() {
+      var self = this;
+      var p1;
+      var newName =  this.$.copyNameInput.value;
+      var newPath = JSON.parse(this.$.copyPathInput.value);
+      var currPath = self.path;
+      if (self.selectType==='dir') {      
+        p1 = self.socket.copyDir({path: currPath, dir: self.selDir, newPath: newPath, newDir: newName});
+        p1.then(function(resp) {
+          self.repopulateList(resp);
+        }, 
+        function(reason) {
+          self.handleError(reason); 
+        });
+      }
+      else if (self.selectType==='key') {
+        p1 = self.socket.copy({path: currPath,key: self.selKey,newPath: newPath, newKey: newName});
+        p1.then(function(resp) {
+            self.repopulateList(resp);
+        },
+        function(reason) {
+          self.handleError(reason); // Error!
+        });
+      }
+    },
+    createNewKey: function() {
+      var self = this;
+      var p1;
+      var newKey = self.$.newKeyInput.value; 
+      var newVal = self.$.newValueInput.value;
+      if (newKey) {
+        p1 = self.socket.set({path: self.path, key: newKey, value: newVal});
+        p1.then(function(resp) {
+          self.repopulateList(resp);
+        }, 
+        function(reason) {
+          self.handleError(reason); // Error!
+        });
+      }
+      else {
+        self.handleError('Cannot create key');
+      }
+    },
+    createNewFolder: function() {
+      var self = this;
+      var p1;
+      var newName = self.$.nameInput.value;
+      if (this.$.nameInput.value){
+        p1 = self.socket.mkDir({path: self.path,dir: newName});
+        p1.then(function(resp) {
+          self.repopulateList(resp);
+        },
+          function(reason) {
+          this.handleError(reason);
+        });
+      }
+      else {
+        this.handleError('Cannot create folder named empty string');      
+     }
+    },
+    updateKey: function(event) {
+      var self = this;
+      var p1;
+      var selKey = Object.keys(event.detail)[0];
+      var newVal = event.detail[selKey]; 
+      p1 = self.socket.set({path: self.path, key: selKey, value: newVal});
+      p1.then(function(resp) {
+        self.repopulateList(resp);
+      }, 
+      function(reason) {
+        self.handleError(reason); // Error!
+      });
+      this.selKey = null;
+      console.log(event.target[0]);
+      event.target.reset();
+      event.target[0].value = null; 
+      //todo fix rendering of input after repopulatelist
+    },
   });
 })();
+
+function clickHandler(e) {
+  //lint complains about this but it's used. This is what makes the dialogs popup
```

Comment:
I'm having trouble getting this to work. It doesn't immediately work if I just put it inside the 
```
function() {
  function clickHandler(e) {
...
```

before (or after) the ```Polymer``` definition